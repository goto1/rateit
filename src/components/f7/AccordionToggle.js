import React from "react";
import PropTypes from "prop-types";

const AccordionToggle = ({ className, children }) => {
  const classNames = `accordion-item-toggle ${className
    ? className
    : ""}`.trim();
  return (
    <div className={classNames}>
      {children}
    </div>
  );
};

AccordionToggle.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired
};

export default AccordionToggle;
