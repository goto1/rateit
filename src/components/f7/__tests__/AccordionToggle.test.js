import React from "react";
import AccordionToggle from "../AccordionToggle";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

describe("AccordionToggle component", () => {
  it("should render correctly", () => {
    const tree = renderer
      .create(
        <AccordionToggle>
          <span>HelloWorld!</span>
        </AccordionToggle>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render a span", () => {
    const wrapper = shallow(
      <AccordionToggle>
        <span>HelloWorld!</span>
      </AccordionToggle>
    );
    const actual = wrapper.find("span").at(0).html();
    const expected = "<span>HelloWorld!</span>";
    expect(actual).toBe(expected);
  });

  it("should render with a specified className", () => {
    const wrapper = shallow(
      <AccordionToggle className="custom">
        <span>HelloWorld!</span>
      </AccordionToggle>
    );
    const actual = wrapper.props().className;
    const expected = "accordion-item-toggle custom";
    expect(actual).toBe(expected);
  });
});
