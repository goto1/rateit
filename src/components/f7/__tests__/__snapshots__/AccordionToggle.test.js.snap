// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccordionToggle component should render correctly 1`] = `


<div
  className="accordion-item-toggle"
>
  <span>
    HelloWorld!
  </span>
</div>
`;
